### PR TITLE
Format on save and update the notification system

### PR DESCRIPTION
**Pull request by joan38**
Solving https://github.com/olafurpg/scalafmt/issues/530

settings

But also moving to the better notification system:
balloon

event log

Cheers

### CODE REVIEW

File: intellij/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
   </change-notes>
 
   <!-- please see http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html for description -->
-  <idea-version since-build="141.0"/>
+  <idea-version since-build="162"/>
```
Comment (olafurpg) [inline intellij/resources/META-INF/plugin.xml]: Any idea how many people still use <2016.2? Unfortunately, I can't tell from the download numbers on plugins.jetbrains.com

**Comment by olafurpg** [comment]
PS. I appreciate you cleaned up the project structure, long overdue.

File: intellij/resources/META-INF/plugin.xml (hunk 1 above)
Comment (joan38) [inline intellij/resources/META-INF/plugin.xml]: Actually I though it was the version it's been built with.
I will revert that then.

**Comment by olafurpg** [comment]
I can't promise when the next release will be, but for the impatient here's a zip file for manual installation: https://github.com/olafurpg/scalafmt/releases/download/v0.4.7/intellij-4.7-autosave.zip

**Comment by rcavalcanti** [comment]
I just tested the zip and indeed works very well.

However, it seems that it enable the format on save behavior to all projects. It's a IDE settings instead of project setting. 

I'm not using scalafmt all over the place, although I plan to do it, I can pretty much imagine that others won't be able to use scalafmt everywhere. 

Having to enable, disable the plugin on a per project base is very annoying and error prone.

**Comment by olafurpg** [comment]
Great observation @rcavalcanti 

One thing we could do is enable format on save only if the setting is enabled globally AND there is a .scalafmt.conf file in the root directory of the current project. How would that work?

**Comment by joan38** [comment]
Or we can have 2 checkboxes, one for project level and the other for global.
And we format if one of the 2 is enabled.

**Comment by olafurpg** [comment]
That sounds good as well. Would you mind taking a swing at this @joan38 ?

**Comment by joan38** [comment]
I can take a java.swing at it over the WE yeah.

**Comment by lunaryorn** [comment]
@joan38 Thanks so much for this change… I feel that I'll be breaking our CI a lot less now 👏 🎆

**Pull request merged**